lib/crowbar/client/command/upgrade/prechecks.rb

```diff
@@ -66,26 +66,32 @@ def content_from(request)
               request.parsed_response["checks"].each do |check_id, values|
                 # make the check_id server agnostic
                 # the check_id could be named differently in the server response
-                check_id = values["errors"].keys.first if values["errors"].any?
+                check_ids = values["errors"].keys if values["errors"].any?
```

Comment:
Well, you're right,  error code is often a number (even dictionaries say so: http://dictionary.cambridge.org/dictionary/english/error-code). It was just a comment. I don't mind you calling it differently than the UI but IMO `check_id` is the key in `checks` hash so this could be e.g. `error_id`.